nxtool/log_providers/elastic.py

```diff
@@ -20,20 +16,39 @@
 except ImportError:  # python3
     from configparser import ConfigParser
 
-from elasticsearch_dsl import Search, Q
+from elasticsearch import TransportError
+from elasticsearch_dsl import Search, Q, Index
+from elasticsearch_dsl import DocType, Date, Boolean, String, Integer, Ip, GeoPoint
 from elasticsearch_dsl.connections import connections
 
 from nxtool.log_providers import LogProvider
 
 
+class Event(DocType):
+    ip = String()   ## Ip() seems not to be supported in old version of elasticsearch
```

Comment:
Old like what?
It's supported in [1.4](https://www.elastic.co/guide/en/elasticsearch/reference/1.4/mapping-ip-type.html)